wordpress-mobile/WordPress-FluxC-Android
# Recover from metadata null key

Summary
==========
Fix the issue https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2499 by allowing the WCMetaData.key field to be nullable. Since Metadata is a data that can be modified at will-be store plugins, we're seeing Gson ignoring Kotlin constraints about nullability and setting a null value within the WCMetaData model.

How to Test
==========
1. Simply ensure that the new test covering the null key scenario is working as expected, or forcefully inject a metadata with a null key inside the `StripOrderMetaData.invoke` call through the orderDto.meta_data parameter and verify it will be removed with success from the function output.

Update release notes:
